### PR TITLE
Restructure documentation and contribution guidelines

### PR DESCRIPTION
**Pull request by alban**
* docs: Add index file and front-matter
* Restructure docs for the hierarchy looks better in the docs portal
* Link to Kinvolk Contribution repo as Authoring PRs
* Add Code of Conduct mention

### CODE REVIEW

**Comment by vbatts** [comment]
a nit, why keep it capitalized and full word "Documentation". I know this is what the kernel does, but this is not consistent across kinvolk projects.
Maybe "docs"?

**Comment by joaquimrocha** [comment]
> a nit, why keep it capitalized and full word "Documentation". I know this is what the kernel does, but this is not consistent across kinvolk projects.
> Maybe "docs"?

I must say I agree. Not sure if I had shared this observation before though, but it'd surely be nice to have `docs` like the rest of the projects.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -124,41 +124,8 @@ $ make integration-tests
...
-   reviewers might need. Use the PR's description to explain why you are
-   proposing the change, give an overview, raise questions about yet-unresolved
-   issues in your PR, list TODO items etc.
-
-PRs which follow these rules are easier to review and merge.
-
-[linux-sep-changes]: https://www.kernel.org/doc/html/v4.17/process/submitting-patches.html#separate-your-changes
-[linux-desc-changes]: https://www.kernel.org/doc/html/v4.17/process/submitting-patches.html#describe-your-changes
+For making PRs/commits consistent and easier to review, please check out
+Kinvolk's [contribution guidelines on git](https://github.com/kinvolk/contribution/tree/master/git).
```
Comment (joaquimrocha) [inline CONTRIBUTING.md:134]: This should rather be:

```suggestion
Kinvolk's [contribution guidelines on git](https://github.com/kinvolk/contribution/blob/master/topics/git.md).
```

**Review by joaquimrocha — CHANGES_REQUESTED** [review]
There is a deprecated link from my old branch.

Also, @alban , we want to have a small description for every section (every `FOLDER/_index.md`), so if you don't mind, please add small intro to what the section is about in
https://github.com/kinvolk/inspektor-gadget/blob/alban_newdocs/Documentation/guides/_index.md

Otherwise readers will just see a listing of the section's article titles and it's not that good.

**Review by joaquimrocha — APPROVED** [review]
Now it LGTM.
I have noticed however that you have CONTRIBUTING.md outside of the `docs` folder. And thus it will not show up in the documentation.

If you add the file into the docs folder then remember to remove the manual index/toc + add the front-matter (the YAML data seen in other files).

**Review by vbatts — APPROVED** [review]
LGTM

**Review by joaquimrocha — APPROVED** [review]

**Pull request merged**